File: app/src/main/java/com/library/filedownload/FileDownloadActivity.java
package com.library.filedownload;

import androidx.annotation.Nullable;

import android.os.Bundle;
import android.view.View;
import android.widget.ProgressBar;
import android.widget.TextView;

import com.blankj.utilcode.util.ToastUtils;
import com.common.base.BaseActivity;
import com.common.util.BitmapUtil;
import com.common.util.filedownload.FileDownloadHelper;
import com.common.util.filedownload.OnDownloadListener;
import com.library.R;

/**
 * 文件下载页面
 */
public class FileDownloadActivity extends BaseActivity {

    private static final String fileUrl = "https://library-collection.oss-cn-beijing.aliyuncs.com/file/file.zip";
    private int ids;
    private ProgressBar progressBar;
    private TextView tvProgress;
    private TextView tvSpeed;
    private FileDownloadHelper fileDownloadHelper;

    @Override
    public int getLayout() {
        return R.layout.activity_file_download;
    }

    @Override
    public void initViewIds() {
        progressBar = findViewById(R.id.progress_bar);
        tvProgress = findViewById(R.id.tv_progress);
        tvSpeed = findViewById(R.id.tv_speed);
    }

    @Override
    public void initView() {

    }

    @Override
    public void initData(@Nullable Bundle savedInstanceState) {

    }

    /**
     * 开始下载
     *
     * @param view
     */
    public void btnStart(View view) {
        startDownload();
    }

    /**
     * 开始下载
     */
    private void startDownload(){
        fileDownloadHelper = new FileDownloadHelper();
        fileDownloadHelper.downloadFile(fileUrl, BitmapUtil.getFileStoragePath(), false, new OnDownloadListener() {

            @Override
            public void onPending(int id, int soFarBytes, int totalBytes) {
            }

            @Override
            public void onProgress(int id, int speed, int soFarBytes, int totalBytes) {
                tvProgress.setText(soFarBytes + " / " + totalBytes);
                tvSpeed.setText(speed + " KB/s");
                progressBar.setMax(totalBytes);
                progressBar.setProgress(soFarBytes);
            }

            @Override
            public void onComplete(String path) {
                ToastUtils.showShort("下载完成");
            }

            @Override
            public void onError(Throwable e) {
                ToastUtils.showShort("下载异常" + e.getMessage());
            }
        });
    }

    /**
     * 暂停下载
     *
     * @param view
     */
    public void btnPause(View view) {
        fileDownloadHelper.pause();
    }
}

File: app/src/main/java/com/library/filedownload/FileDownloadActivity.kt
package com.library.filedownload

import android.os.Bundle
import android.view.View
import android.widget.ProgressBar
import android.widget.TextView
import com.blankj.utilcode.util.AppUtils
import com.blankj.utilcode.util.ToastUtils
import com.common.base.BaseActivity
import com.common.http.helper.filedownload.FileDownloadHelper
import com.common.http.helper.filedownload.OnDownloadListener
import com.common.util.BitmapUtil.fileStoragePath
import com.library.R
import kotlinx.android.synthetic.main.activity_file_download.*

/**
 * 文件下载页面
 */
class FileDownloadActivity : BaseActivity() {
    private var fileDownloadHelper: FileDownloadHelper? = null
    override val layout: Int
        get() = R.layout.activity_file_download

    override fun initViewIds() {
    }

    override fun initView() {}
    override fun initData(savedInstanceState: Bundle?) {}
    /**
     * 开始下载
     *
     * @param view
     */
    fun btnStart(view: View?) {
        startDownload()
    }

    /**
     * 开始下载
     */
    private fun startDownload() {
        fileDownloadHelper = FileDownloadHelper()
        fileDownloadHelper!!.downloadFile(fileUrl, fileStoragePath, false, object : OnDownloadListener {
            override fun onPending(id: Int, soFarBytes: Int, totalBytes: Int) {}
            override fun onProgress(id: Int, speed: Int, soFarBytes: Int, totalBytes: Int) {
                tv_progress!!.text = "$soFarBytes / $totalBytes"
                tv_speed!!.text = "$speed KB/s"
                progress_bar!!.max = totalBytes
                progress_bar!!.progress = soFarBytes
            }

            override fun onComplete(path: String?) {
                ToastUtils.showShort("下载完成")
                //安装App
                AppUtils.installApp(path)
            }

            override fun onError(e: Throwable?) {
                ToastUtils.showShort("下载异常" + e!!.message)
            }
        })
    }

    /**
     * 暂停下载
     *
     * @param view
     */
    fun btnPause(view: View?) {
        fileDownloadHelper!!.pause()
    }

    companion object {
        private const val fileUrl = "https://library-collection.oss-cn-beijing.aliyuncs.com/file/file.apk"
    }
}